src/cr/cube/measures/pairwise_pvalues.py

```diff
@@ -18,61 +18,67 @@
         return self._intersperse_insertions_rows_and_columns(
-            1.0 - WishartCDF(self._pairwise_chisq, self._n_min, self._n_max).values
+            1.0 - WishartCDF(props, self._n_min, self._n_max).values
         )
 
+    @staticmethod
+    def _factory(slice_, axis, weighted):
+        if slice_.dim_types[0] == DT.MR_SUBVAR:
+            return _MrXCatPairwiseSignificance(slice_, axis, weighted)
+        return _CatXCatPairwiseSignificance(slice_, axis, weighted)
+
     @lazyproperty
-    def _categorical_pairwise_chisq(self):
-        """Pairwise comparisons (Chi-Square) along axis, as numpy.ndarray.
+    def _insertions_indices(self):
+        """Return H&S indices of the pairwise comparison dimension."""
+        return self._slice.inserted_hs_indices()[1 - self._axis]
 
-        Returns a square, symmetric matrix of test statistics for the null
-        hypothesis that each vector along *axis* is equal to each other.
-        """
-        chisq = np.zeros([self._numel, self._numel])
-        for i in xrange(1, self._numel):
-            for j in xrange(0, self._numel - 1):
-                chisq[i, j] = chisq[j, i] = np.sum(
-                    np.square(self._proportions[:, i] - self._proportions[:, j])
-                    / self._observed
-                ) / (1 / self._margin[i] + 1 / self._margin[j])
+    def _intersperse_insertions_rows_and_columns(self, pairwise_pvals):
+        """Return pvals matrix with inserted NaN rows and columns, as numpy.ndarray.
 
-        return chisq
+        Insertions (Headers and Subtotals) create offset in calculated pvals, and
+        these need to be taken into account, when converting them to columnar letters
+        representation. For this reason, we need to insert an all-NaN row and a
+        column in the right indices (the inserted indices of the H&S, in the
+        respective dimension).
+        """
+        for i in self._insertions_indices:
+            pairwise_pvals = np.insert(pairwise_pvals, i, np.nan, axis=0)
+            pairwise_pvals = np.insert(pairwise_pvals, i, np.nan, axis=1)
+        return pairwise_pvals
 
     @lazyproperty
     def _margin(self):
         """Margin for the axis as numpy.ndarray."""
         return self._slice.margin(axis=self._axis)
 
     @lazyproperty
-    def _off_margin(self):
-        """Margin for the opposite axis as numpy.ndarray."""
-        return self._slice.margin(axis=(1 - self._axis))
-
-    @lazyproperty
-    def _pairwise_chisq(self):
-        """Pairwise Chi-squared statistics along axis, as numpy.ndarray.
-
-        Zscore is a measure of statistical significance of observed vs.
-        expected counts. It's only applicable to a 2D contingency tables.
-        """
-        return self._categorical_pairwise_chisq
-
-    @lazyproperty
-    def _proportions(self):
-        """Slice proportions for *axis* as numpy.ndarray."""
-        return self._slice.proportions(axis=self._axis)
+    def _numel(self):
```

Comment:
I would strongly prefer `._element_count` here. What is a numel? Sounds like it could be something I don't know about (but isn't). This name will be used outside this property; best to make it communicate clearly without having to look down here to understand what it means.

Reply:
Will do. This is a remnant of the initial implementation...